.ci/build-directories.cake

```diff
@@ -0,0 +1,366 @@
+#addin nuget:?package=Cake.XCode&version=4.1.0
+#addin nuget:?package=Cake.Json&version=4.0.0&loadDependencies=true
+#addin nuget:?package=redth.xunit.resultwriter&version=1.0.0
+
+
+// SECTION: Arguments and Settings
+
+var DEFAULT_BUILD_TARGET = "ci";
+var POD_UPDATE_PREFIXES = new [] { "iOS/", "Mac/", "XPlat/" };
+
+var VERBOSITY = Argument("v", Argument("verbosity", Verbosity.Normal));
+var CONFIGURATION = Argument("c", Argument("configuration", "Release"));
+
+var GIT_PREVIOUS_COMMIT = Argument("gitpreviouscommit", "");
+var GIT_COMMIT = Argument("gitcommit", "");
+var GIT_BRANCH = Argument("gitbranch", "origin/master");
+var GIT_EXE = Argument("gitexe", "git");
+
+var ROOT_DIR = MakeAbsolute((DirectoryPath)Argument("root", "."));
+
+var ROOT_OUTPUT_DIR = ROOT_DIR.Combine("output");
+var COPY_OUTPUT_TO_ROOT = Argument("copyoutputtoroot", true);
+
+var BUILD_NAMES = Argument("names", Argument("name", Argument("n", "")))
+	.Split(new [] { ",", ";" }, StringSplitOptions.RemoveEmptyEntries);
+var BUILD_TARGETS = Argument("targets", Argument("build-targets", Argument("build-targets", Argument("build", DEFAULT_BUILD_TARGET))))
+	.Split(new [] { ",", ";" }, StringSplitOptions.RemoveEmptyEntries);
+
+var FORCE_BUILD = Argument("force", Argument("forcebuild", Argument("force-build", false)));
+var POD_REPO_UPDATE = Argument("update", Argument("repo-update", Argument("pod-repo-update", false)));
+
+var BUILD_GROUPS = GetFiles($"{ROOT_DIR}/*/**/build.cake")
```

Comment:
Instead of the yaml file, our script is based on these build.cake files. As a result the component "name" is now the path to the folder with the cake.

For example, the "Kotlin" component is now "Android/Kotlin". It always builds on both Windows and macOS, so the build script has to decide wether or not to run anything.